build.sc

```diff
@@ -1,139 +1,211 @@
-import mill._, scalalib._, publish._, scalajslib.ScalaJSModule
+
+import mill._, scalalib._, publish._, scalajslib._, scalanativelib._, mill.scalanativelib.api._
 val sjsonnetVersion = "0.2.7"
 
-object sjsonnet extends Cross[SjsonnetModule]("2.12.12", "2.13.3")
-class SjsonnetModule(val crossScalaVersion: String) extends Module {
-  def millSourcePath = super.millSourcePath / ammonite.ops.up
-  trait SjsonnetCrossModule extends CrossScalaModule with PublishModule{
+val crossScalaVersions = Seq("2.13.3", "2.12.12")
 
-    def artifactName = "sjsonnet"
-    def platformSegment: String
+object sjsonnet extends Module{
+  
+  object native extends Cross[SjsonnetNativeModule]("2.11.12")
+  class SjsonnetNativeModule(val crossScalaVersion: String) extends SjsonnetCrossModule with CommonJvmNative with ScalaNativeModule{
+    def releaseMode = ReleaseMode.ReleaseFast
+    def nativeLinkStubs = true
+    def libcrypto = T(os.walk(os.root / "usr").filter(_.last == "libcrypto.a").head.toString)
+    def staticLinkLibcrypto = T { true }
+    def libcryptoLinkingOptions = T {
+      if(staticLinkLibcrypto()) {
+        Array(libcrypto())
+      } else {
+        val libcryptoDir = os.Path(libcrypto()) / os.up
+        Array(s"-L$libcryptoDir", "-lcrypto")
+      }
+    }
```

Comment:
This can be removed if we allow only dynamic linking, this means that users will need to install openssl to use sjsonnet.

Reply:
@lolgab maybe let's just stub out the MD5 implementation with `???` for now, I think that's what we're already doing on the JS side of things